src/plot.jl

```diff
@@ -0,0 +1,157 @@
+represented by the thickness of the graph edges, as well as displayed values.
+"""
+function plot_graph(graph)
+    nodelabelcolor = RGB(0.345, 0.164, 0.376)
+    transportflowcolor = RGB(1, 0.647, 0)
+    assetflowcolor = RGB(0.988, 0.525, 0.110)
+    nodecolor = RGBA(0.345, 0.164, 0.376, 0.5)
+
+    node_labels = labels(graph) |> collect
+
+    temp_graph = DiGraph(nv(graph))
+    for e in edges(graph)
+        add_edge!(temp_graph, e.src, e.dst)
+        if graph[node_names[e.src], node_names[e.dst]].is_transport
+            add_edge!(temp_graph, e.dst, e.src)
+        end
+    end
+
+    node_size = []
+    node_names = []
+    for a in node_labels
+        node_total_capacity =
+            graph[a].initial_capacity +
+            graph[a].investable * graph[a].investment * graph[a].capacity
+        push!(node_names, "$a \n $(node_total_capacity)") # "Node Name \n Capacity: ##"
+        push!(node_size, node_total_capacity)
+    end
+    node_size = node_size * 98 / maximum(node_size) .+ 5
+
+    edge_colors = []
+    edge_width = []
+    edge_labels = []
+    for e in edges(temp_graph)
+        from = node_names[e.src]
+        to = node_names[e.dst]
+        edge_data = has_edge(graph, e.src, e.dst) ? graph[from, to] : graph[to, from]
+
+        if edge_data.is_transport
+            ttc =
+                (
+                    if has_edge(graph, e.src, e.dst)
+                        edge_data.initial_export_capacity
+                    else
+                        edge_data.initial_import_capacity
+                    end
+                ) + edge_data.investable * edge_data.investment * edge_data.capacity
+            push!(edge_labels, string(ttc))
+            push!(edge_width, ttc)
+            push!(edge_colors, transportflowcolor)
+        else
+            push!(edge_colors, assetflowcolor)
+            push!(edge_labels, "")
+            push!(edge_width, 0)
+        end
+    end
+    edge_width = edge_width * 0.9 / (maximum(edge_width) == 0 ? 1 : maximum(edge_width)) .+ 0.1 # Normalize edge_width with minimum of 0.1 (just visible)
+
+    #TODO Show initial vs investment somehow (couldn't get node stroke to work)
+    f, ax, p = graphplot(
+        temp_graph;
+        node_size = node_size,
+        node_strokewidth = 0,
+        node_color = nodecolor,
+        ilabels = node_names,
+        ilabels_color = nodelabelcolor,
+        ilabels_fontsize = 10.0,
+        edge_width = edge_width * 10,
+        arrow_size = (edge_width .+ 2) * 10,
+        edge_color = edge_colors,
+        elabels = edge_labels,
+        elabels_fontsize = 10.0,
+        elabels_color = edge_colors,
+    )
+
+    #TODO Make these axis adjustments a calculation (so labels aren't cut off)
+    CairoMakie.xlims!(ax, -5.5, 5.5)
+    CairoMakie.ylims!(ax, -5, 6)
+    hidedecorations!(ax)
+    hidespines!(ax)
+
+    return f
+end
+
+function plot_graph(energy_problem)
+    plot_graph(energy_problem.graph)
+end
```

Comment:
```suggestion
function plot_graph(energy_problem::EnergyProblem)
    plot_graph(energy_problem.graph)
end
```
You have to specialize the type here, because the function already exists with a more general type on the top.
The same for all the other functions.
My initial comment on removing the types are for the docstrings - that being said, the only place that you really need to define the types are for the energy_model variables as far as I can tell.

Reply:
Ah right! Good catch. :)